src/installer/pkg/nuget/Microsoft.NETCore.DotNetAppHost/Microsoft.NETCore.DotNetAppHost.proj

```diff
@@ -0,0 +1,15 @@
+<Project Sdk="Microsoft.Build.NoTargets">
+  <PropertyGroup>
+    <!-- Reference the outputs for the dependency nodes calculation. -->
+    <NoTargetsDoNotReferenceOutputAssemblies>false</NoTargetsDoNotReferenceOutputAssemblies>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <!-- Listing the runtime specific packages to populate the dependencies section.
+         Not building these references to avoid unintentional Build/Pack invocations.
+         They are included in host-packages.proj based on the OutputRID. -->
+    <ProjectReference Include="runtime.*.$(MSBuildProjectName).proj"
```

Comment:
@ViktorHofer does this seem reasonable? This is based on the runtime.native.System.IO.Ports projects.

I wanted to just have one `runtime.Microsoft.NETCore.DotNetAppHost.proj` that could be referenced with different properties (each supported RID) and set its package ID based on the output RID, but I couldn't figure out how to make them actually get recognized as separate dependencies in here

Reply:
Not sure if that would work as NuGet uses the project.assets.json information to calculate package dependencies. cc @ericstj and @carlossanlop while I'm out.

Reply:
Yeah, I think that's what I hit - the project.assets.json just had the one reference to the project (with the first RID) and so the generated nuspec just had the one dependency.

Reply:
The downside with the current approach is that it is static and new RIDs can't be built for that package without a file addition. So not really dynamic via msbuild properties / items.

Maybe what you tried might actually work, by having multiple P2Ps to that single runtime project but with different RuntimeIdentifier properties passed in? Unsure what exactly is encoded in the project.assets.json file. Trying that out should be easy.

Reply:
If you can transform some RID list to ProjectReferences that set different global properties passed into the same project and that global property is used to specify the PackageID you can represent all the dependencies.  Might depend on how much msbuild customization you can do while still participating in restore.

Reply:
Here's what makes NuGet drop duplicate P2Ps with different metadata:

<img width="1130" alt="image" src="https://github.com/dotnet/runtime/assets/7412651/0141e9a4-dcdb-4862-ac62-d8e190166732">

Reply:
With the following changes, the msbuild graph items are calculated correctly (unsure about static graph restore though, need to test): https://github.com/NuGet/NuGet.Client/compare/dev...ViktorHofer:NuGet.Client:ProjectReferencesWithDifferentPropertiesInDGSpec

Still, the dgspec.json file still doesn't contain the ridlib dependencies with the different projectNames. The [`WriteRestoreGraphTask` msbuild task](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Build.Tasks/WriteRestoreGraphTask.cs) is responsible for creating the .dg file which is then read and parsed to retrieve the project nodes.

That task uses the `ProjectUniqueName` metadata as the id in a dictionary: https://github.com/NuGet/NuGet.Client/blob/0dd5a1ea536201af94725353e4bc711d7560b246/src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/MSBuildRestoreUtility.cs#L48C8-L57

ProjectUniqueName is set to `MSBuildProjectFullPath` in NuGet.targets, i.e. here: https://github.com/NuGet/NuGet.Client/blob/0dd5a1ea536201af94725353e4bc711d7560b246/src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets#L680

Because of that, NuGet treats project nodes with the same path identically when writing the .dg/.dgspec.json file. I wonder if there's a better key, i.e. a combination of the project full path and the project name?

<img width="402" alt="image" src="https://github.com/dotnet/runtime/assets/7412651/6076c9c0-94f3-48c4-be5a-00c001964a73">

cc @nkolev92 @zivkan @jeffkl

Reply:
With [one more commit that updates the project unique name](https://github.com/ViktorHofer/NuGet.Client/commit/beabd2933fb2f651a68607b4e7db3794cee890e7), the dg.json file now finally contains the additional P2Ps: https://gist.github.com/ViktorHofer/948c9bf2ccf8dec8030d4c4ef61e7a20 (look at the project key, unique name and project name keys).

Unfortunately, that's still not enough as NuGet now fails to resolve the ridLib project:

> NU1105: Unable to find project information for 'C:\Users\vihofer\Downloads\ridLib\ridLib\ridLib.csproj'.

Reply:
There's lots going on so I'm not 100% sure I can provide appropriate help. 

These are project references to projects with a physical project file?
And all project refs have the same assembly name.
Can you change "PackageId" to something unique? Or do you pack all of these independently?

Reply:
I will move this discussion into a NuGet issue and start an offline conversation with you. I think this is just a scenario that NuGet currently doesn't support, but not intentionally.